### PR TITLE
Improve ArcCoordinates interface

### PR DESCRIPTION
**Pull request by johannes-fischer**
Adds another constructor, `__eq__` and `__repr__` functions to `ArcCoordinates` to make it more usable.

### CODE REVIEW

File: lanelet2_python/python_api/geometry.cpp [hunk 1]
```diff
@@ -289,8 +290,21 @@ BOOST_PYTHON_MODULE(PYTHON_API_MODULE_NAME) {  // NOLINT
   def("area", boost::geometry::area<ConstHybridPolygon2d>);
 
   class_<ArcCoordinates>("ArcCoordinates", "Coordinates along an arc", init<>())
+      .def("__init__", make_constructor(
+                           +[](double length, double distance) {
+                             return boost::make_shared<ArcCoordinates>(ArcCoordinates{length, distance});
+                           },
+                           default_call_policies(),
+                           (arg("length") = 0., arg("distance") = 0.)))
```
Comment (poggenhans) [inline lanelet2_python/python_api/geometry.cpp:298]: The default constructor is now is now superfluous with the new constructor. And boost also understands std::shared_ptr:
```suggestion
  class_<ArcCoordinates>("ArcCoordinates", "Coordinates along an arc", no_init)
      .def("__init__", make_constructor(
                           +[](double length, double distance) {
                             return std::make_shared<ArcCoordinates>(ArcCoordinates{length, distance});
                           },
                           default_call_policies(),
                           (arg("length") = 0., arg("distance") = 0.)))
```

**Review by poggenhans — CHANGES_REQUESTED** [review]
Thanks for the all the PRs! Looks like the CI is currently broken, I'm fixing this with #184

File: lanelet2_python/python_api/geometry.cpp (hunk 1 above)
Comment (johannes-fischer) [reply lanelet2_python/python_api/geometry.cpp:298]: Sounds good, I am not very experienced with boost, just trying to understand and copy from the existing interface code.

**Comment by johannes-fischer** [comment]
I first had moved the `ArcCoordinates` interface definition from `python_api/geometry.cpp` to `python_api/core.cpp` in https://github.com/fzi-forschungszentrum-informatik/Lanelet2/pull/351/commits/3e8ffc419fc5185b8c74de8916871ecbd8e35ff4, because in the c++ code `ArcCoordinates` is defined in `lanelet2_core/primitives/Point.h` and generally it seems that stuff from `lanelet2_core/primitives` is put into `python_api/core.cpp` whereas `lanelet2_core/geometry` goes into `python_api/geometry.cpp`.
But since this might a breaking change in the python api I reverted it. Do you have an opinion on this?

**Comment by poggenhans** [comment]
> I first had moved the ArcCoordinates interface definition from python_api/geometry.cpp to python_api/core.cpp

Do you see a good reason to move it? Other than for symmetry to C++? Imho the organization of things in python doesn't need to match the C++ organization. In C++ there are a few more constrains that won't allow us to organize things in the way we can do it in python (e.g. by using a separate geometry module). And since `ArcCoordinates` only makes sense in combination with the geometry module, I chose to put it there.

**Review by poggenhans — APPROVED** [review]

**Comment by johannes-fischer** [comment]
The symmetry to C++ was my only reason.

**Pull request merged**